lib/auth/auth.go

```diff
@@ -1396,18 +1396,19 @@ func (a *AuthServer) CreateAccessRequest(req services.AccessRequest) error {
 	return trace.Wrap(err)
 }
 
-func (a *AuthServer) SetAccessRequestState(reqID string, state services.RequestState, updatedBy ...string) error {
-	if err := a.DynamicAccess.SetAccessRequestState(reqID, state); err != nil {
+func (a *AuthServer) SetAccessRequestState(ctx context.Context, reqID string, state services.RequestState) error {
+	if err := a.DynamicAccess.SetAccessRequestState(ctx, reqID, state); err != nil {
 		return trace.Wrap(err)
 	}
-	u := "unknown"
-	if len(updatedBy) == 1 {
-		u = updatedBy[0]
+	updateBy, ok := ctx.Value(events.AccessRequestUpdateBy).(string)
+	if !ok {
+		log.Warnf("access-request state updated by unknown party: %q", reqID)
```

Comment:
Log lines should be full sentences. So `Access-request state updated by unknown party: %q."`

Reply:
Also, shouldn't the request fail here? What scenario would the request be updated but the updater be unknown?

Reply:
This is a side-effect of moving the audit event emission from `AuthWithRoles` to `AuthServer`.  Since `AuthServer` may be called *internally*, it isn't guaranteed that the method will be invoked in the context of a specific user RPC call.

My understanding is that the reason why the audit event emission was moved from `AuthWithRoles` to `AuthServer` was precisely so that the emission of audit events was decoupled from the expectation of a user RPC context.

All that being said, the only time this method is *currently* called outside of a user RPC context is during tests, so it wouldn't be too troublesome tweak the tests and make updater a required value.

Reply:
I would make it a requirement and update tests. We don't want "unknown" to slip into production audit messages, would make attribution difficult.